compiler/rustc_privacy/src/lib.rs

```diff
@@ -476,16 +496,17 @@ impl<'tcx> EmbargoVisitor<'tcx> {
             return;
         }
 
-        if self.get(local_def_id).is_none() {
+        if self.effective_visibilities.public_at_level(local_def_id).is_none() {
             return;
         }
 
         // Since we are starting from an externally visible module,
         // all the parents in the loop below are also guaranteed to be modules.
         let mut module_def_id = macro_module_def_id;
+        let macro_ev = self.get(local_def_id);
```

Comment:
Could you add an assert `assert!(macro_ev.is_some())`?
The condition above should make the none case impossible.